ui/src/api/tree/query/variables.ts

```diff
@@ -27,9 +27,17 @@ const fetchVariables = async ({
     },
   });
 
-export const useWorkflowVariables = ({ path }: { path: string }) => {
+export const useWorkflowVariables = ({
+  path,
+  namespace: givenNamespace,
+}: {
+  path: string;
+  namespace?: string | null;
+}) => {
   const apiKey = useApiKey();
-  const namespace = useNamespace();
+  const defaultNamespace = useNamespace();
+
+  const namespace = givenNamespace ? givenNamespace : defaultNamespace;
```

Comment:
Should we use `givenNamespace ?? defaultNamespace` here as well? I think it reads a bit nicer.

Just to point out the difference again:

The shorter version of `givenNamespace ? givenNamespace : defaultNamespace;` is actually `givenNamespace || defaultNamespace`. But I would prefer `givenNamespace ?? defaultNamespace`. The key difference is, that we also want to use the `defaultNamespace` when `givenNamespace` is an empty string. 

Same applies to `ui/src/api/variables/query/useVariables.ts`